### PR TITLE
Fix: messageActions do not re-render when change pinned status

### PR DESCRIPTION
**Pull request by huuthinh245**
## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) spec
- [ ] New code is covered by tests
- [ ] Screenshots added for visual changes
- [ ] Documentation is updated

### CODE REVIEW

File: package/src/components/Message/MessageSimple/MessageContent.tsx [hunk 1]
```diff
@@ -453,7 +453,8 @@ const areEqual = <
     prevMessage.reply_count === nextMessage.reply_count &&
     prevMessage.status === nextMessage.status &&
     prevMessage.type === nextMessage.type &&
-    prevMessage.text === nextMessage.text;
+    prevMessage.text === nextMessage.text &&
+    prevMessage?.pinned === nextMessage?.pinned;
```
Comment (madsroskar) [inline package/src/components/Message/MessageSimple/MessageContent.tsx:457]:
```suggestion
    prevMessage.pinned === nextMessage.pinned;
```
It doesn't seem like optional chaining would be necessary here, mind removing it? Or if I'm wrong, why is it needed here and not in other uses of `prevMessage`?

Comment (huuthinh245) [reply package/src/components/Message/MessageSimple/MessageContent.tsx:457]: Screen Shot 2021-11-30 at 10 47 38
I saw the varible is optional. so Make sure every thing is working i do like this

**Comment by huuthinh245** [comment]
updated

**Comment by madsroskar** [comment]
Could you change the commit messages to follow the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)? For example, if you use the git CLI you could reset the two commits and make a new commit with the changes of both like this:

1. `git reset --soft HEAD~2`
2. `git commit -m "fix: Rerender message actions when message.pinned changes"`
3. `git push origin develop --force-with-lease`

We also need for the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) to be signed before we can merge the PR.

**Comment by huuthinh245** [comment]
Hi sir. please check again. 1st create pull request, so i have some mistake, forgive me

**Review by khushal87 — COMMENTED** [review]
Pinned message feature is not yet released for `develop`. It will be released with `v4.0.0`. can you explain how are to trying to render it in the `develop` branch if it's not yet released?

**Comment by huuthinh245** [comment]
My project is using pin action. I found the the bug when user pinned  on android,  message actions do not re render. So i have  to try fix the bug. I hope this help for everyone. if it's not yet released. It's so okay. i just want notify for you the bug

**Comment by khushal87** [comment]
Thanks, @huuthinh245 for clearing this. :)

**Comment by madsroskar** [comment]
> My project is using pin action. I found the the bug when user pinned on android, message actions do not re render it keep pin on UI instead of unpin . So i have to try fix the bug. I hope this help for everyone. if it's not yet released. It's so okay. i just want notify for you the bug

And thanks a lot for doing so, it's helpful for us to hear what we can do to make implementing the Chat into your application easier, and to fix things like this where the memoization needs to be aware of the property on message to understand that the object is updated. We really appreciate this!

It turns out that we do have this check as part of v4.0.0 that we're releasing a beta version of this week, probably on Thursday. I'll therefore close this PR for now, and I'll give you an update here once the beta release of v4.0.0 is ready so you can test it out in your app if you would want to.

**Comment by huuthinh245** [comment]
final How can i build from source code. I can't find anything document for it. I need to fix some bug and build. We can't wait maintainer release new version

**Comment by madsroskar** [comment]
That's something you'll have to do with `yarn` or `npm` for example, as that is the responsibility of your package manager. However, since the exposed package `stream-chat-react-native` is in a subdirectory, I don't think you'll be able to use the git remote path directly since it can't take subdirectories into account.

It might for now work better to create a tarball for your version of `package/native-package` that you can link to as `stream-chat-react-native`, and one for `package/` that you link to as `stream-chat-react-native-core`. See for example [`git archive`](https://git-scm.com/docs/git-archive) to make the tarballs from your local clone of the fork you're using.

We are planning on making this simpler to do in the future, but for now I think releasing your fork on the npm registry might be easiest. I don't know yet when we'll be able to make this simpler, but I'll make sure to bump the task in our backlog a bit up the list.

You can for example see the documentation of `npm` as a reference [here](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#git-urls-as-dependencies)

**Comment by huuthinh245** [comment]
Thanks for reply, I have a question about  license. If I publish on npm. Do I have to request  permission  about it or something like this.

**Comment by vishalnarkhede** [comment]
Hey @huuthinh245 if you plan to publish on npm, then please publish it as a private package.

**Pull request closed without merging**